applications/FluidDynamicsApplication/custom_elements/surface_smoothing_element.cpp

```diff
@@ -0,0 +1,670 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Mohammad R. Hashemi
+//
+
+// System includes
+
+// External includes
+
+// Include Base headers
+
+// Project includes
+#include "custom_elements/surface_smoothing_element.h"
+
+namespace Kratos
+{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+    constexpr static std::array<unsigned int, 4> mNode0ID = {2, 0, 0, 0};
+    constexpr static std::array<unsigned int, 4> mNode1ID = {3, 3, 1, 2};
+    constexpr static std::array<unsigned int, 4> mNode2ID = {1, 2, 3, 1};
+
+    constexpr static unsigned int num_dim  = 3;
```

Comment:
This is only valid for 3D cases. All the new elements in the FluidDynamicsApplications must be 2D and 3D compatible. What we normally do is to template the class with the dimensions and number of nodes.

Reply:
After discussions with @RiccardoRossi I think it is better to merge at this stage (keeping everything only for 3D). Implementing all routines (besides this process) also for 2D is not possible at the moment.

Reply:
Which are the features that we lack to be 2D compatible?

Reply:
To be honest, I have never tested my process/solver for 2D. 
For this special one (smoothing), I _think_ both the element and process should be templated with dimension (to keep track of sizes), and edges be created instead of faces to implement BC.

Reply:
I think so (similar to what we did for the other level set processes). Then should be more or less "easy" to add the dimension template right?

Reply:
I will give it a try for this process.

Reply:
Cool. Thanks.

Reply:
Done.